File: 2025/regional-andaluza/I/code/cpp/I.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int beauty(int n, const vector<int>& beauty1, const vector<int>& beauty2) {
    if (n == 1) {
        return max(beauty1[0], beauty2[0]);
    }

    vector<int> dp1(1, beauty1[0]);
    vector<int> dp2(1, beauty2[0]);

    if (n > 1) {
        dp1.push_back(dp2[0] + beauty1[1]);
        dp2.push_back(dp1[0] + beauty2[1]);
    }

    for (int i = 2; i < n; ++i) {
        dp1.push_back(beauty1[i] + max(dp2[i - 1], dp2[i - 2]));
        dp2.push_back(beauty2[i] + max(dp1[i - 1], dp1[i - 2]));
    }

    return max(dp1[n - 1], dp2[n - 1]);
}

int main() {
    int t;
    cin >> t;  // Read number of test cases
    while (t--) {
        int n;
        cin >> n;  // Read the value of n
        vector<int> list1(n), list2(n);

        // Read first list of beauty values
        for (int i = 0; i < n; ++i) {
            cin >> list1[i];
        }

        // Read second list of beauty values
        for (int i = 0; i < n; ++i) {
            cin >> list2[i];
        }

        // Call beauty function and print the result
        cout << beauty(n, list1, list2) << endl;
    }

    return 0;
}

File: 2025/regional-andaluza/I/code/py/I.py
def beauty(n, beauty1, beauty2):
    if n == 1:
        return max(beauty1[0], beauty2[0])
    
    dp1 = [beauty1[0]]
    dp2 = [beauty2[0]]
    
    if n > 1:
        dp1.append(dp2[0] + beauty1[1])
        dp2.append(dp1[0] + beauty2[1])
    
    for i in range(2, n):
        dp1.append(beauty1[i] + max(dp2[i-1], dp2[i-2]))
        dp2.append(beauty2[i] + max(dp1[i-1], dp1[i-2]))
    
    return max(dp1[n-1], dp2[n-1])


if __name__ == "__main__":
    t = int(input())  # Read number of test cases
    for _ in range(t):
        n = int(input())  # Read the value of n
        list1 = list(map(int, input().split()))  # Read first list of beauty values
        list2 = list(map(int, input().split()))  # Read second list of beauty values
        
        result = beauty(n, list1, list2)
        print(result)
